xdsl/transforms/printf_to_putchar.py

```diff
@@ -0,0 +1,242 @@
+            scf.Yield.get(new_integer, digits)
+
+        return while_loop.results[1]
+
+    # print_minus_if_negative implementation
+    def print_minus_if_negative(integer: SSAValue) -> scf.If:
+        """
+        This function returns an MLIR scf.if op that prints a minus sign
+        if the integer value is negative and returns the absolute value
+        of the integer in either case, in python code:
+        ```
+        if(integer > 0):
+            print("-")
+            return (0 - integer)
+        else:
+            return integer
+        ```
+        """
+        is_negative_block = Block()
+        is_positive_block = Block()
+        is_negative = arith.Cmpi.get(integer, zero, "slt")
+
+        # Either way get the absolute value of the number
+        absolute_value = scf.If.get(
+            is_negative, [i32], [is_negative_block], [is_positive_block]
+        )
+        # If the value is negative, print the minus sign, return abs value
+        with ImplicitBuilder(is_negative_block):
+            PrintCharOp.from_constant_char("-")
+            negative_input = arith.Subi(zero, integer)
+            scf.Yield.get(negative_input)
+        # If the value is positive, just return value itself as abs value
+        with ImplicitBuilder(is_positive_block):
+            scf.Yield.get(integer)
+
+        return absolute_value
+
+    def print_digits(digits: SSAValue, absolute_value: SSAValue) -> scf.For:
+        """
+        Return an scf.for loop that prints all digits of the given value.
+        In python code:
+        for i in range(0, digits, 1):
+            position = (digits - 1) - i
+            digit = (absolute_value // (10**position)) % 10
+            print(digit)
+        """
+        zero_index = arith.Constant.from_int_and_width(0, IndexType())
+        one_index = arith.Constant.from_int_and_width(1, IndexType())
+        digits_index = arith.IndexCastOp((digits,), (IndexType(),))
+        loop_body = Block(arg_types=(IndexType(),))
+
+        # Print all from most significant to least
+        for_loop = scf.For.get(zero_index, digits_index, one_index, (), loop_body)
+        with ImplicitBuilder(loop_body) as (index_var,):
+            one = arith.Constant.from_int_and_width(1, i32)
+            size_minus_one = arith.Subi(digits, one)
+            index_var_int = arith.IndexCastOp((index_var,), (i32,))
+            position = arith.Subi(size_minus_one, index_var_int)
+            # digit = (num // (10**pos)) % 10
+            ten = arith.Constant.from_int_and_width(10, i32)
+            i_0 = math.IPowIOp((ten, position), (i32,))
+            i_1 = arith.DivUI(absolute_value, i_0)
+            digit = arith.RemUI(i_1, ten)
+            # ascii value for zero is 48 in decimal
+            ascii_offset = arith.Constant.from_int_and_width(48, i32)
+            char = arith.Addi(digit, ascii_offset)
+            char_i8 = arith.TruncIOp.get(char, IntegerType(8))
```

Comment:
`nit`:  you could probably define again an alias `i8` at the top of the file, right ?